python_modules/libraries/dagster-k8s/dagster_k8s/job.py

```diff
@@ -446,6 +447,28 @@ def config_type_run_launcher(cls):
                     description="Raw Kubernetes configuration for launched runs.",
                 ),
                 "job_namespace": Field(StringSource, is_required=False, default_value="default"),
+                "allowed_user_defined_k8s_config_fields": Field(
+                    Shape(
+                        {
+                            "container_config": Field(Map(key_type=str, inner_type=bool)),
+                            "pod_spec_config": Field(Map(key_type=str, inner_type=bool)),
+                            "job_metadata": Field(Map(key_type=str, inner_type=bool)),
+                            "job_spec_config": Field(Map(key_type=str, inner_type=bool)),
+                            "namespace": Field(bool),
+                        }
+                    ),
+                    is_required=False,
+                    description="Dictionary of fields that are allowed to be configured on a "
+                    "per-run or per-code-location basis - e.g. using tags on the run. "
+                    "Can be used to prevent user code from being able to set arbitrary kubernetes "
+                    "config on the pods launched by the run launcher.",
+                ),
+                "allowed_user_defined_env_vars": Field(
+                    Array(str),
+                    is_required=False,
+                    description="List of environment variable names that are allowed to be set on "
+                    "a per-run or per-code-location basis - e.g. using tags on the run. ",
+                ),
```

Comment:
how exactly does this work when its not filled out / partially filled out?

Reply:
nothing filled out for this field at all - allow everything (like today)
partially filled out - only allow the things that are filled out

i.e. if you set it to 
```
container_config:
  env
```

you can ONLY set config that results in env being customized (you can't set anything that would change pod_spec_config)

Reply:
The overall absence -> everything permitted, subsection omission -> denied bit feels a little goofy to me. Maybe `only_allow_` instead of `allowed_` naming helps?

This `Map[str, bool]` structure prevents controlling anything more deeply nested, is that OK? Maybe would need to like a list of field paths instead if more flexibility was needed. 

I see the inner config doesn't default anything so you end up needing to fill all the top level out to set anything.